tests/countries/test_turks_and_caicos_islands.py

```diff
@@ -0,0 +1,240 @@
+#  holidays
+#  --------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: Vacanza Team and individual contributors (see CONTRIBUTORS file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+from unittest import TestCase
+
+from holidays.countries.turks_and_caicos_islands import TurksAndCaicosIslands, TC, TCA
+from tests.common import CommonCountryTests
+
+
+class TestTC(CommonCountryTests, TestCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass(TurksAndCaicosIslands, years=range(1963, 2050))
+
+    def test_country_aliases(self):
+        self.assertAliases(TurksAndCaicosIslands, TC, TCA)
+
+    def test_no_holidays(self):
+        self.assertNoHolidays(TurksAndCaicosIslands(years=1962))
+
+    def test_new_years_day(self):
+        self.assertHolidayName("New Year's Day", (f"{year}-01-01" for year in range(1963, 2050)))
+
```

Comment:
For holidays that have "observed" feature, we need to test observed dates (look at Trinidad and Tobago, for example).